packages/gatsby-link/src/index.tsx

```diff
@@ -1,12 +1,48 @@
 import PropTypes from "prop-types"
 import React from "react"
-import { Link } from "@reach/router"
+import {
+  Link,
+  LinkProps,
+  LinkGetProps,
+  NavigateFn,
+  NavigateOptions,
+} from "@reach/router"
 
 import { parsePath } from "./parse-path"
 
 export { parsePath }
 
-export function withPrefix(path) {
+export interface IGatsbyLinkState {
+  IOSupported: boolean
+}
+
+// eslint-disable-next-line @typescript-eslint/interface-name-prefix
```

Comment:
why are we doing this eslint disable? let's just name it `IGatsbyLinkProps`

Reply:
`GatsbyLinkProps` was previously exported in index.d.ts, so this was my attempt at keeping the new generated index.d.ts compatible.

Alternatively, I can name this IGatsbyLinkProps and export a deprecated typedef called GatsbyLinkProps? or what do you think?